Bluetooth: Mesh: fix proxy solicitation

alxelax opened the pull request:
PR fixes mesh proxy solicitation functionality as well as bluetooth tester regarding it.

alkh-local@alkh-linux:~/auto-pts$ python3 autoptsclient-zephyr.py nordic -i 192.168.56.101 192.168.56.101 -l 192.168.56.1 192.168.56.1  -t /dev/ttyACM0 -b nrf52 -c MESH/SR/MPXS/B -S 65000 65002 -C 65001 65003
(139872873231408) Starting PTS 192.168.56.101:65000 ...
(139872873227472) Starting PTS 192.168.56.101:65002 ...
(139872873231408) OK
(139872873227472) OK
 1/12   MESH   MESH/SR/MPXS/BV-01-C   PASS           32.248                  
 2/12   MESH   MESH/SR/MPXS/BV-02-C   PASS           28.548                  
 3/12   MESH   MESH/SR/MPXS/BV-03-C   PASS           31.956                  
 4/12   MESH   MESH/SR/MPXS/BV-04-C   PASS           28.183                  
 5/12   MESH   MESH/SR/MPXS/BV-05-C   PASS           30.147                  
 6/12   MESH   MESH/SR/MPXS/BV-06-C   PASS           28.102                  
 7/12   MESH   MESH/SR/MPXS/BV-07-C   PASS           30.66                  
 8/12   MESH   MESH/SR/MPXS/BV-09-C   PASS           87.456                  
 9/12   MESH   MESH/SR/MPXS/BV-10-C   PASS           28.518                  
10/12   MESH   MESH/SR/MPXS/BV-11-C   PASS           31.457                  
11/12   MESH   MESH/SR/MPXS/BV-12-C   PASS           96.635                  
12/12   MESH   MESH/SR/MPXS/BV-13-C   PASS           28.508                  

Fixes #63861

alxelax committed:
Bluetooth: Mesh: fix proxy solicitation

Commit makes workable proxy solicitation functionality
only server part without dependencies on client part.
Only on demand proxy server is required.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
tests: bluetooth: tester: enable proxy solicitation

Commit enables proxy solicitation functionality
(server part) for mesh proxy PTS tests.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

jhedberg commented on subsys/bluetooth/mesh/solicitation.c:
```diff
@@ -190,13 +190,14 @@ void bt_mesh_sol_recv(struct net_buf_simple *buf, uint8_t uuid_list_len)
 	type = net_buf_simple_pull_u8(buf);
 	if (type != BT_DATA_UUID16_SOME && type != BT_DATA_UUID16_ALL) {
-		LOG_ERR("Invalid type 0x%x, expected 0x%x or 0x%x",
+		LOG_DBG("Invalid type 0x%x, expected 0x%x or 0x%x",
```
Elsewhere in the host I think we've tried to use WARN log level for potentially invalid external data, whereas ERR is reserved for local failures. Is there a good reason to use DBG instead of WARN here (and elsewhere in the file)?

jhedberg replied:
Correction, WRN level (we used to have `BT_WARN()` but now it's `LOG_WRN()`.

alxelax replied:
Actually, it shouldn't happen because it is pre-parsed here: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/adv.c#L338-L346
IMHO this checking is redundant at all. However, this is bugfix not feature refactoring.

jhedberg replied:
Sure, but I don't think the log level changes are needed for the fix, right? They could be in a different commit or PR.

alxelax replied:
They are very useful. Right now, without added debug messages, mesh device just silently ignores everything what PTS toolchain sends to it even I enable debugging of proxy, models etc.
If you insist, I'll add them as a separate commit in this PR. The maintenance and bugfixing of proxy solicitation functionality are much complex without them.

alxelax committed:
Bluetooth: Mesh: improve solicitation debug logging

Commits adds debug logging for all reasons
preventing of the proxy solicitation.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>
